File: haskell/src/Types/Genero.hs
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE DeriveAnyClass #-}

module Types.Genero(Genero(..)) where

import GHC.Generics (Generic)
import Data.Aeson (ToJSON, FromJSON)

data Genero
  = Rock
  | Pop
  | Eletronica
  | HipHop
  | Rap
  | Funk
  | MPB
  | Sertanejo
  | Forro
  | Indie
  | Pagode 
  deriving (Show, Eq, Ord, Enum, Read, Bounded, Generic, ToJSON, FromJSON)

File: src/Types/Genero.pl
:- module(genero, [genero/1, generos/1]).

genero(rock).
genero(pop).
genero(eletronica).
genero(classico).
genero(hiphop).
genero(rap).
genero(funk).
genero(mpb).
genero(sertanejo).
genero(forro).
genero(indie).
genero(pagode).
genero(reggae).

generos(Lista) :- findall(G, genero(G), Lista).
